src/BetterReflection.php

```diff
@@ -84,7 +94,9 @@ public function phpParser() : Parser
     public function astLocator() : AstLocator
     {
         return $this->astLocator
-            ?? $this->astLocator = new AstLocator($this->phpParser());
+            ?? $this->astLocator = new AstLocator($this->phpParser(), function () : FunctionReflector {
+                return $this->functionReflector();
```

Comment:
doesn't that mean you're using a different function reflector each time? Looks a bit weird to me and it also makes it way harder to instantiate the different Reflectors without `BetterReflection` (e.g. in PHP-Scoper since I'm configuring the locators differently

Reply:
Yes, I don't like it either. I used this hack as temporary solution to solve circular dependency. I would be very happy for better suggestion how to solve it.

Reply:
This is only inside the `BetterReflection` instance, which is acceptable to me.

Reply:
Err yeah it's just that I don't see a nice way to do it for me:

```php
f (null === $this->reflector) {
            $phpParser = $this->getParser();
            $astLocator = new Locator($phpParser);

            $sourceLocator = new MemoizingSourceLocator(
                new PhpInternalSourceLocator(
                    $astLocator,
                    new AggregateSourceStubber(
                        new PhpStormStubsSourceStubber($phpParser),
                        new ReflectionSourceStubber()
                    )
                )
            );

            $classReflector = new ClassReflector($sourceLocator);

            $functionReflector = new FunctionReflector($sourceLocator, $classReflector);

            return new Reflector(
                $classReflector,
                $functionReflector,
                new ConstantReflector($sourceLocator, $classReflector)
            );
        }

        return $this->reflector;
```

Btw the locator is marked as `@internal` but I don't really see a convenient way to do without for now...